src/helpers/helpers.canvas.js

```diff
@@ -3,10 +3,11 @@ import {clampAll} from './helpers.core';
 import {calculateTextAlignment, getSize} from './helpers.options';
 
 const widthCache = new Map();
+const isDOMSupported = typeof window !== 'undefined' && typeof document !== 'undefined';
 
-export function isImageOrCanvas(content) {
-  return content instanceof Image || content instanceof HTMLCanvasElement;
-}
+export const isImageOrCanvas = isDOMSupported
+  ? (content) => content instanceof Image || content instanceof HTMLCanvasElement
+  : (content) => content instanceof Image;
```

Comment:
I have some doubts that `instanceof` might not work properly in every situation. With shadow DOM, frames etc. Also the `Image` class might not be available in every canvas implementation for node.

So I think your first implementation was actually closer to what we should have.

```js
/**
 * Determine if content is an image or a canvas.
 * @param {*} content
 * @returns boolean|undefined
 * @todo move this function to chart.js helpers
 */
export function isImageOrCanvas(content) {
  if (content && typeof content === 'object') {
    const type = content.toString();
    return (type === '[object HTMLImageElement]' || type === '[object HTMLCanvasElement]');
  }
}
```